actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -89,11 +92,20 @@ class ActionCable.Connection
 
     open: ->
       ActionCable.log("WebSocket onopen event")
-      @disconnected = false
-      @subscriptions.reload()
+      # This is in the case that the browser client wants
+      # to open the connection regardless of whether a
+      # protocol was negotiated successfully.
+      if @webSocket?.protocol not in ActionCable.PROTOCOLS
+        @close()
+      else
+        @disconnected = false
+        @subscriptions.reload()
 
-    close: ->
+    close: (event) ->
       ActionCable.log("WebSocket onclose event")
+      if @webSocket?.protocol not in ActionCable.PROTOCOLS
+        ActionCable.log("WebSocket is closing because there is no common protocol. Stopping ConnectionMonitor.")
+        @monitor.stop()        
```

Comment:
Yeah, I think that'd work fine. Would be cool to hot-reload the client-side js, but that's probably more trouble than it's worth for what will be an infrequent event.